From 651410fa2a2c9c1e390ecbe384ea259f9bd319c8 Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <cgarcia@igalia.com>
Date: Thu, 24 Feb 2011 17:38:40 +0100
Subject: [PATCH] Copy event axes for double/triple click events in gdk_event_copy()

Button event axes for double/triple click events are freed by
gdk_event_free(), so copy them to avoid an invalid free when releasing a
copied event.
---
 gdk/gdkevents.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/gdk/gdkevents.c b/gdk/gdkevents.c
index 5188e1e..25853da 100644
--- a/gdk/gdkevents.c
+++ b/gdk/gdkevents.c
@@ -555,6 +555,8 @@ gdk_event_copy (const GdkEvent *event)
       break;
 
     case GDK_BUTTON_PRESS:
+    case GDK_2BUTTON_PRESS:
+    case GDK_3BUTTON_PRESS:
     case GDK_BUTTON_RELEASE:
       if (event->button.axes)
         new_event->button.axes = g_memdup (event->button.axes,
-- 
1.7.4.1

